tdp/cli/commands/plan/edit.py

```diff
@@ -0,0 +1,220 @@
+# - Add extra variables using the `with` keyword and separate with `,`.
+#   Example: wait_sleep with rolling-interval=5, mock-arg=example
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+"""
+
+
+def _parse_line(line: str) -> tuple[str, Optional[str], Optional[list[str]]]:
+    """Parses a line which contains an operation, and eventually a host and extra vars.
+
+    Args:
+        line: line to be parsed
+
+    Returns:
+        Operation, host and extra vars
+    """
+    parsed_line = re.match(
+        r"^(.*?)( on .*?){0,1}( ?with .*?){0,1}( ?on .*?){0,1}$", line
+    )
+
+    if parsed_line is None:
+        raise ValueError(
+            "Error on line '"
+            + line
+            + "' : it must be 'OPERATION [on HOST] [with EXTRA_VARS[,EXTRA_VARS]].'"
+        )
+
+    if parsed_line.group(1).split(" ")[0] == "":
+        raise ValueError("Error on line '" + line + "' : it must contain an operation.")
+
+    if len(parsed_line.group(1).strip().split(" ")) > 1:
+        raise ValueError("Error on line '" + line + "' : only 1 operation is allowed.")
+
+    if parsed_line.group(2) is not None and parsed_line.group(4) is not None:
+        raise ValueError(
+            "Error on line '" + line + "' : only 1 host is allowed in a line."
+        )
+
+    operation = parsed_line.group(1).strip()
+
+    # Get the host and test if it is declared
+    if parsed_line.group(2) is not None:
+        host = parsed_line.group(2).split(" on ")[1]
+        if host == "":
+            raise ValueError(
+                "Error on line '" + line + "' : host is required after 'on' keyword."
+            )
+    elif parsed_line.group(4) is not None:
+        host = parsed_line.group(4).split(" on ")[1]
+        if host == "":
+            raise ValueError(
+                "Error on line '" + line + "' : host is required after 'on' keyword."
+            )
+    else:
+        host = None
+
+    # Get the extra vars and test if they are declared
+    if parsed_line.group(3) is not None:
+        extra_vars = parsed_line.group(3).split(" with ")[1]
+        if extra_vars == "":
+            raise ValueError("Extra vars are required after 'with' keyword.")
+        extra_vars = extra_vars.split(",")
+        extra_vars = [item.strip() for item in extra_vars]
+    else:
+        extra_vars = None
+
+    return (operation, host, extra_vars)
+
+
+def _parse_file(file_name) -> list[tuple[str, Optional[str], Optional[list[str]]]]:
+    """Parses a file which contains operations, hosts and extra vars."""
+    file_content = file_name.read()
+    return [
+        _parse_line(line)
+        for line in file_content.split("\n")
+        if line and not line.startswith("#")
+    ]
+
+
+@contextmanager
+def _managed_temp_file(**kwargs):
+    """Creates a temporary file and deletes it when the context is exited.
+
+    Args:
+        **kwargs: arguments to be passed to tempfile.NamedTemporaryFile
```

Comment:
"arguments" to "Arguments" and "." at the and

Reply:
Done